lib/runtime-core/src/vm.rs

```diff
@@ -716,10 +716,19 @@ impl LocalGlobal {
 }
 
 /// Identifier for a function signature.
+///
+/// A transparent `SigIndex`
 #[derive(Debug, Clone, Copy)]
 #[repr(transparent)]
 pub struct SigId(pub u32);
```

Comment:
Should we replace all uses of `SigId` with `SigIndex` ?

Reply:
The fact that it's `transparent` may be important to its use.  I just added the comment because I get confused every time I see it.

Reply:
Is there any way to verify that we are making use of `transparent` (kind of: tests without it will pass?)

If tests pass, then we can just remove this type and use SigIndex directly.

Reply:
It's stored by pointer and might be accessed by the generator code, having it not be transparent could mean that depending on the optimization level, for example, it might just corrupt the data that generated code sees.  It is possible to have tests for this, but it's probably not the simplest thing to test.  I don't really know how it's used, so maybe it's easier to test than I think it is.